src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -101,20 +101,24 @@ private UserDto mapUserToDto(User user) {
         return convertToDto(parent.getChildren(), ChildDto.class);
     }
 
-    public void addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
+    public List<ChildDto> addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
         final List<Child> childrenToSave = new ArrayList<>(convertAllTo(
                 childrenDtoToAdd, Child.class, ArrayList::new));
         addNewChildren(parent, childrenToSave);
         userRepository.save(parent);
+        return convertToDto(findDuplicatedChildren(parent.getChildren(), childrenToSave), ChildDto.class);
```

Comment:
You could do it without extra methods and much more simpler:
```
        List<Child> children = parent.getChildren();
        children.retainAll(childrenToSave);
        return convertToDto(children, ChildDto.class);
```

Reply:
Performed